specs/OAuth/AuthManager.md

```diff
@@ -0,0 +1,891 @@
+
+        // Additional parameters set by the authorization server in the response URI.
+        Windows.Foundation.Collections.IMapView<String, String> AdditionalParams { get; };
+    }
+
+    [contract(OAuthContract, 1)]
+    runtimeclass AuthFailure
+    {
+        // From the "error" parameter of the error response. The value of this property will map to a well known string
+        // specified in RFC 6749 sections 4.1.2.1 and 4.2.2.1, or approved extensions.
+        //
+        // Defined by RFC 6749: The OAuth 2.0 Authorization Framework, sections 4.1.2.1 and 4.2.2.1
+        //      https://www.rfc-editor.org/rfc/rfc6749#section-4.1.2.1
+        //      https://www.rfc-editor.org/rfc/rfc6749#section-4.2.2.1
+        String Error { get; };
+
+        // From the "error_description" parameter of the error response. An optional parameter that, when set, provides
+        // additional human-readable information intended to assist the developer in understanding the error.
+        //
+        // Defined by RFC 6749: The OAuth 2.0 Authorization Framework, sections 4.1.2.1 and 4.2.2.1
+        //      https://www.rfc-editor.org/rfc/rfc6749#section-4.1.2.1
+        //      https://www.rfc-editor.org/rfc/rfc6749#section-4.2.2.1
+        String ErrorDescription { get; };
+
+        // From the "error_uri" parameter of the error response. An optional parameter that, when set, specifies a URI
+        // identifying a human-readable webpage intended to assist the developer in understanding the error.
+        //
+        // Defined by RFC 6749: The OAuth 2.0 Authorization Framework, sections 4.1.2.1 and 4.2.2.1
+        //      https://www.rfc-editor.org/rfc/rfc6749#section-4.1.2.1
+        //      https://www.rfc-editor.org/rfc/rfc6749#section-4.2.2.1
+        Windows.Foundation.Uri ErrorUri { get; };
+
+        // From the "state" parameter of the error response. This property will always be set because a state value is
+        // always sent with the request.
+        //
+        // Defined by RFC 6749: The OAuth 2.0 Authorization Framework, sections 4.1.2.1 and 4.2.2.1
+        //      https://www.rfc-editor.org/rfc/rfc6749#section-4.1.2.1
+        //      https://www.rfc-editor.org/rfc/rfc6749#section-4.2.2.1
+        String State { get; };
+
+        // Additional parameters set by the authorization server in the response URI.
+        Windows.Foundation.Collections.IMapView<String, String> AdditionalParams { get; };
+    }
+
+    [contract(OAuthContract, 1)]
+    runtimeclass AuthRequestResult
+    {
+        // The raw URI that was used to complete the request.
+        Windows.Foundation.Uri ResponseUri { get; };
+
+        // Non-null if the server's response indicates success, otherwise null
+        AuthResponse Response { get; };
+
+        // Non-null if the server's response indicates failure, otherwise null
+        AuthFailure Failure { get; };
+    }
+
+    [contract(OAuthContract, 1)]
+    runtimeclass TokenRequestParams
```

Comment:
Since this is a v1 of the effort, what is the minimum-bar that an app needs to use OAuth2 correctly? I see all the places you are supporting the RFC, but what is the most common use case? Can we do less now, see if it meets @dotMorten 's needs, and add more later?

Reply:
Based on our discussion, 
Adding the below two APIs for simplification and keeping the existing ones for a comprehensive list. 
1. InitiateAuthRequestAsync(Uri fullAuthorizeURI, Uri RedirectURI)
2. InitiateAuthRequestAsyncWithoutCallback(Uri fullAuthorizeUri)